qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java

```diff
@@ -0,0 +1,581 @@
+          .withEnv("IDENTITY_RETRY_ATTEMPTS", "90")
+          .withEnv("IDENTITY_RETRY_DELAY_SECONDS", "1")
+          .withEnv("IDENTITY_DATABASE_HOST", DATABASE_HOST)
+          .withEnv("IDENTITY_DATABASE_PORT", String.valueOf(DATABASE_PORT))
+          .withEnv("IDENTITY_DATABASE_NAME", DATABASE_NAME)
+          .withEnv("IDENTITY_DATABASE_USERNAME", DATABASE_USER)
+          .withEnv("IDENTITY_DATABASE_PASSWORD", DATABASE_PASSWORD)
+          .withNetwork(NETWORK)
+          .withExposedPorts(8080, 8082)
+          .waitingFor(
+              new HttpWaitStrategy()
+                  .forPort(8082)
+                  .forPath("/actuator/health")
+                  .allowInsecure()
+                  .forStatusCode(200))
+          .withNetworkAliases("identity");
+
+  @AutoCloseResource private static final TestStandaloneBroker ZEEBE = new TestStandaloneBroker();
+
+  @SuppressWarnings("unused")
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher =
+      new ContainerLogsDumper(
+          () -> Map.of("postgres", POSTGRES, "keycloak", KEYCLOAK, "identity", IDENTITY));
+
+  private BpmnModelInstance process;
+
+  @BeforeAll
+  static void init() throws Exception {
+    ZEEBE
+        .withGatewayConfig(
+            gateway -> {
+              gateway.getMultiTenancy().setEnabled(true);
+              gateway.getSecurity().getAuthentication().setMode(AuthMode.IDENTITY);
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setBaseUrl(
+                      "http://%s:%d".formatted(IDENTITY.getHost(), IDENTITY.getMappedPort(8080)));
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setIssuerBackendUrl(
+                      "http://%s:%d%s"
+                          .formatted(
+                              KEYCLOAK.getHost(),
+                              KEYCLOAK.getMappedPort(8080),
+                              KEYCLOAK_PATH_CAMUNDA_REALM));
+              gateway
+                  .getSecurity()
+                  .getAuthentication()
+                  .getIdentity()
+                  .setAudience(ZEEBE_CLIENT_AUDIENCE);
+            })
+        .start()
+        .await(TestHealthProbe.READY);
+
+    awaitCamundaRealmAvailabilityOnKeycloak();
+
+    associateTenantsWithClient(List.of("tenant-a"), ZEEBE_CLIENT_ID_TENANT_A);
+    associateTenantsWithClient(List.of("tenant-b"), ZEEBE_CLIENT_ID_TENANT_B);
+    associateTenantsWithClient(List.of("tenant-a", "tenant-b"), ZEEBE_CLIENT_ID_TENANT_A_AND_B);
+  }
+
+  @BeforeEach
+  void setup() {
+    process =
+        Bpmn.createExecutableProcess(Strings.newRandomValidBpmnId())
+            .startEvent()
+            .serviceTask("task", b -> b.zeebeJobType("type"))
+            .endEvent()
+            .done();
+  }
+
+  @Test
+  void shouldDenyDeployProcessWhenUnauthorized() {
```

Comment:
:wrench: It might be good to add another test case where a process is deployed to multiple tenants and the process version is independently incremented.